cfme/middleware/server.py

```diff
@@ -59,25 +67,19 @@ def __init__(self, name, provider=None, **kwargs):
         self.provider = provider
         self.product = kwargs['product'] if 'product' in kwargs else None
         self.id = kwargs['id'] if 'id' in kwargs else None
+        self.path = kwargs['path'] if 'path' in kwargs else None
 
     @classmethod
     def servers(cls, provider=None):
         servers = []
-        if provider:
-            # if provider instance is provided try to navigate provider's servers page
-            # if no servers are registered in provider, returns empty list
-            if not provider.load_all_provider_servers():
-                return []
-        else:
-            # if provider instance is not provided then navigates  to all servers page
-            sel.force_navigate('middleware_servers')
+        _get_servers_page(provider=provider)
         if sel.is_displayed(list_tbl):
-            for page in paginator.pages():
+            for _ in paginator.pages():
                 for row in list_tbl.rows():
                     servers.append(MiddlewareServer(name=row.server_name.text,
                                                     id=row.feed.text,
                                                     product=row.product.text,
-                                                    provider=row.provider.text))
+                                                    provider=provider))
```

Comment:
@hhovsepy we may not construct provider easily, because it needs more details(`ip`,`port`, etc.,). We may introduce in properties to hold these values.